src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,246 @@
+        assertThat(testHash.containsAll(extendHash)).isFalse();
+        testHash.remove(3112);
+        testHash.remove(511);
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isFalse();
+
+        /* Test: Check the Map against itself */
+        testHash = extendHash;
+        assertThat(extendHash.containsAll(testHash)).isTrue();
+        assertThat(testHash.containsAll(extendHash)).isTrue();
+    }
+
+    @Test
+    void add() {
+        testHash = new HashSet<>();
+        extendHash = new HashSet<>();
+
+        /* Test: Adding only one Integer each time */
+        assertThat(extendHash).isEmpty();
+        assertThat(extendHash.hashCode()).isEqualTo(0);
+        for (int i = 1; i <= 25; i++) {
+            extendHash.add(i);
+        }
+        assertThat(extendHash).isNotEmpty();
+        assertThat(extendHash.size()).isEqualTo(25);
+        assertThat(extendHash.hashCode()).isEqualTo(325);
+
+        /* Test: Adding more than one at once */
+        assertThat(testHash).isEmpty();
+        assertThat(testHash.hashCode()).isEqualTo(0);
+        testHash.addAll(extendHash);
+        assertThat(testHash).isNotEmpty();
+        assertThat(testHash.size()).isEqualTo(25);
+        assertThat(testHash.hashCode()).isEqualTo(325);
+        assertThat(testHash).isEqualTo(extendHash);
+
+        /* Test: Adding something that already exist (nothing is then happening) */
+        testHash.add(25);
+        assertThat(testHash).isEqualTo(extendHash);
+    }
+
+    @Test
+    void remove() {
+        testPrep();
+        testHash.addAll(extendHash);
+
+        /* Test: Removing only one Integer each time */
+        testHash.remove(13);
+        assertThat(testHash.hashCode()).isEqualTo(extendHash.hashCode() - 13);
+        testHash.remove(77);
+        assertThat(testHash.hashCode()).isEqualTo(extendHash.hashCode() - 77 - 13);
+        testHash.remove(7);
+        assertThat(testHash.hashCode()).isEqualTo(extendHash.hashCode() - 7 - 77 - 13);
+
+        /* Test: Removing more than one at once */
+        extendHash.removeAll(testHash);
+        assertThat(extendHash.hashCode()).isEqualTo(7 + 13 + 77);
+
+        /* Test: Removing something that not exist (nothing is then happening) */
+        testHash = new HashSet<>(extendHash);
+        testHash.remove(2);
+        assertThat(testHash).isEqualTo(extendHash);
+    }
+
+    @Test
+    void clear() {
+        testPrep();
+        extendHash.clear();
+        assertThat(extendHash).isEmpty();
+        assertThat(extendHash).isNotNull();
+    }
+
+    /**
+     * Sets the test variable to the starting point.
+     */
+    void testPrep() {
+        testHash = new HashSet<>();
+        extendHash = new HashSet<>();
+        extendHash.add(511);
+        extendHash.add(2106);
+        extendHash.add(1310);
+        extendHash.add(2401);
+        for (int i = 0; i < 100; i++) {
+            extendHash.add(i);
+        }
+    }
+}
```

Comment:
Die Methode `HashSet#clone()` wurde nicht getestet.

Reply:
Weil ich diese Fehlermeldung bekomme, dass es nicht public ist.
clone()' in 'edu.hm.hafner.util.SetTest' clashes with 'clone()' in 'java.lang.Object'; attempting to assign weaker access privileges ('package-private'); was 'protected'
Wie kann ich das umgehen?

Reply:
Du hast die Testmethode 'clone' genannt. Wie du vielleicht weißt, erbt jede Klasse von der Klasse `Object`. Dort ist eine Methode `clone` definiert, die du dadurch überschreibst. Beim Überschreiben einer Methode darf der Zugriff durch den verwendeten access modifier nur gelockert aber nicht eingeschränkt werden. Deshalb kommt diese Fehlermeldung. Da du die Methode `clone` in der Klasse `Object` aber gar nicht überschreiben willst, verwende einfach einen anderen Methodennamen. Am besten bei Namen für Testmethoden immer einen Prefix wie 'should' oder 'test' verwenden. Also zum Beispiel 'shouldClone'.